SproxydKeysScan/Logging.js

```diff
@@ -0,0 +1,110 @@
+const werelogs = require('werelogs');
+const { env } = require('./env');
+
+/**
+ * @class
+ * @classdesc - aggregates events in a map and logs them at an interval.
+ * call .run() after instantiation to being loop.
+ */
+class AggregateLogger {
+    /**
+     * @constructor
+     * @param {number} logInterval - log summary statistics once every intervalSize seconds
+     */
+    constructor(logInterval) {
+        this.log = new werelogs.Logger('s3utils:SproxydKeysScan:AggregateLogger');
+        if (AggregateLogger._instance) {
```

Comment:
Why is this a singleton?

Reply:
to explicitly enforce one to many w.r.t. aggregatorLogger to ProxyLoggers. When we make a new proxy wereloggers.Logger instance we have to check if an existing AggregateLogger exists and this was one way of doing so

Reply:
This enforces 1 to many w.r.t. AggregateLogger to proxy loggers. So, when a new a proxy logger is created, this is a way to enforce it uses the existing instance and we don't get disjoint summary logs.

Reply:
Oh I see what is going on, I was expecting the use case to be:
```js
const agg = new AggregateLogger(...);
const aLogger = new Logger(... whatever logger );
agg.Add(aLogger);
```

or something similar, the way it is now is sort of like, you create both logggers and the aggregate and they just kinda "link" together magically. Kind of confusing to me.